lib/Parser/Parser.cpp

```diff
@@ -508,8 +526,25 @@ bool Parser::typeCheckInst(Inst::Kind IK, unsigned &Width,
     return false;
   }
 
-  if (!typeCheckOpsMatchingWidths(OpsMatchingWidths, ErrStr))
-    return false;
+  // Note : We don't 'typeCheckOpsMatchingWidths' because the
+  // overflow instructions operands and the inst width is different
+  // The overflow instruction is a tuple of two elements (i32, i1)
+  // that makes it width (32 + 1 = 33). Likewise, for 64-bit operands,
+  // it will be (64 + 1 = 65).
```

Comment:
Nit: redundant space before colon, missing dot after "is different", "that makes it width" could be rephrased.